xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -375,33 +386,24 @@ protected Helper delegate() {
 
       // This is triggered by child balancer
       @Override
-      public void updateBalancingState(ConnectivityState newState, SubchannelPicker newPicker) {
+      public void updateBalancingState(ConnectivityState newState,
+          final SubchannelPicker newPicker) {
         checkNotNull(newState, "newState");
         checkNotNull(newPicker, "newPicker");
 
-        class LoadRecordPicker extends SubchannelPicker {
-          private final SubchannelPicker delegate;
-
-          private LoadRecordPicker(SubchannelPicker delegate) {
-            this.delegate = delegate;
-          }
-
-          @Override
-          public PickResult pickSubchannel(PickSubchannelArgs args) {
-            return statsStore.interceptPickResult(delegate.pickSubchannel(args), locality);
-          }
-
-          @Override
-          public String toString() {
-            return MoreObjects.toStringHelper(this)
-                .add("delegate", delegate)
-                .add("locality", locality)
-                .toString();
-          }
-        }
-
         currentChildState = newState;
-        currentChildPicker = new LoadRecordPicker(newPicker);
+        currentChildPicker =
+            new SubchannelPicker() {
+              SubchannelPicker instance =
```

Comment:
All wrap pickers should override `toString()` otherwise the log will lose picker information. Maybe it's time to define a `ForwardingSubchannelPick` that implements `toString()` like https://github.com/grpc/grpc-java/blob/v1.21.0/core/src/main/java/io/grpc/internal/ForwardingManagedChannel.java#L93

Reply:
It only has one effective method in terms of behavior. For now, I will just put `toString()`.